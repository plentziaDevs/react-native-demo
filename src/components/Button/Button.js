import { string } from 'prop-types';
import React, { PureComponent } from 'react';
import { TouchableWithoutFeedback, View, Text } from 'react-native';

import styles from './Button.style';

class Button extends PureComponent {
  static defaultProps = {
    title: string.isRequired,
  };

  render() {
    const { props: { title, ...inherit } } = this;

    return (
      <TouchableWithoutFeedback {...inherit}>
        <View style={[styles.container, inherit.style]}>
          <Text style={styles.title}>{title}</Text>
        </View>
      </TouchableWithoutFeedback>
    );
  }
}

export default Button;
