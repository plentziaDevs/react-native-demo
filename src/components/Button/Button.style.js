import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: 'black',
    borderRadius: 20,
    height: 40,
    justifyContent: 'center',
    marginVertical: 5,
    width: '100%',
  },

  title: {
    color: 'white',
  },
});
