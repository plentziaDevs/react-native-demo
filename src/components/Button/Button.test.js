import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('<Button>', () => {
  it('renders', () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

