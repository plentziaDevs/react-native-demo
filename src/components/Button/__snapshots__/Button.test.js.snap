// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "black",
        "borderRadius": 20,
        "height": 40,
        "justifyContent": "center",
        "marginVertical": 5,
        "width": "100%",
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
      }
    }
  />
</View>
`;
